### PR TITLE
[Kernel] Fix input for flashinfer prefill wrapper.

### LINKED ISSUE #362: RuntimeError: Out of workspace memory in AlignedAlloactor when there is a lot of GPU memory left
Hi, I encountered an out of workspace memory error when trying to load the gemma-2-27b model using vllm with the flashinfer backend, which seems to have come from flashinfer. I printed out the GPU memory usage list when this happened (see below), and there was still a lot of free memory across the four GPUs, so I'm not sure what the problem is. I'd appreciate any advice on how to fix this issue. Thanks! 

Minimal amount of code to reproduce this error: 
```python
from vllm import LLM
import flashinfer
import os

os.environ['VLLM_ATTENTION_BACKEND'] = 'FLASHINFER'
model = LLM(model="google/gemma-2-27b-it",
                    dtype="auto",
                    trust_remote_code=True,
                    tokenizer_mode="auto",
                    tensor_parallel_size=4)
```

Full log:
```
(VLLM pid=1950315) WARNING 07-07 13:43:20 utils.py:562] Gemma 2 uses sliding window attention for every odd layer, which is currently not supported by vLLM. Disabling sliding window and capping the max length to the sliding window size (4096).
(VLLM pid=1950315) INFO 07-07 13:43:20 config.py:698] Defaulting to use mp for distributed inference
(VLLM pid=1950315) INFO 07-07 13:43:20 llm_engine.py:169] Initializing an LLM engine (v0.5.1) with config: model='google/gemma-2-27b-it', speculative_config=None, tokenizer='google/gemma-2-27b-it', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, rope_scaling=None, rope_theta=None, tokenizer_revision=None, trust_remote_code=True, dtype=torch.bfloat16, max_seq_len=4096, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=4, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=False, kv_cache_dtype=auto, quantization_param_path=None, device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='outlines'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None), seed=0, served_model_name=google/gemma-2-27b-it, use_v2_block_manager=False, enable_prefix_caching=False)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) INFO 07-07 13:43:21 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) WARNING 07-07 13:43:21 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) INFO 07-07 13:43:21 multiproc_worker_utils.py:215] Worker ready; awaiting tasks
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) INFO 07-07 13:43:21 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) WARNING 07-07 13:43:21 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) INFO 07-07 13:43:21 multiproc_worker_utils.py:215] Worker ready; awaiting tasks
(VLLM pid=1950315) INFO 07-07 13:43:21 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) INFO 07-07 13:43:21 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) WARNING 07-07 13:43:21 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) WARNING 07-07 13:43:21 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) INFO 07-07 13:43:21 multiproc_worker_utils.py:215] Worker ready; awaiting tasks
(VLLM pid=1950315) INFO 07-07 13:43:21 utils.py:741] Found nccl from library libnccl.so.2
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) INFO 07-07 13:43:21 utils.py:741] Found nccl from library libnccl.so.2
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) INFO 07-07 13:43:21 utils.py:741] Found nccl from library libnccl.so.2
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) INFO 07-07 13:43:21 utils.py:741] Found nccl from library libnccl.so.2
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) INFO 07-07 13:43:21 pynccl.py:63] vLLM is using nccl==2.20.5
(VLLM pid=1950315) INFO 07-07 13:43:21 pynccl.py:63] vLLM is using nccl==2.20.5
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) INFO 07-07 13:43:21 pynccl.py:63] vLLM is using nccl==2.20.5
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) INFO 07-07 13:43:21 pynccl.py:63] vLLM is using nccl==2.20.5
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) WARNING 07-07 13:43:21 custom_all_reduce.py:118] Custom allreduce is disabled because it's not supported on more than two PCIe-only GPUs. To silence this warning, specify disable_custom_all_reduce=True explicitly.
(VLLM pid=1950315) WARNING 07-07 13:43:21 custom_all_reduce.py:118] Custom allreduce is disabled because it's not supported on more than two PCIe-only GPUs. To silence this warning, specify disable_custom_all_reduce=True explicitly.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) WARNING 07-07 13:43:21 custom_all_reduce.py:118] Custom allreduce is disabled because it's not supported on more than two PCIe-only GPUs. To silence this warning, specify disable_custom_all_reduce=True explicitly.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) WARNING 07-07 13:43:21 custom_all_reduce.py:118] Custom allreduce is disabled because it's not supported on more than two PCIe-only GPUs. To silence this warning, specify disable_custom_all_reduce=True explicitly.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) INFO 07-07 13:43:22 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) WARNING 07-07 13:43:22 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) INFO 07-07 13:43:22 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) INFO 07-07 13:43:22 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) WARNING 07-07 13:43:22 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) WARNING 07-07 13:43:22 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) INFO 07-07 13:43:22 selector.py:79] Using Flashinfer backend.
(VLLM pid=1950315) WARNING 07-07 13:43:22 selector.py:80] Flashinfer will be stuck on llama-2-7b, please avoid using Flashinfer as the backend when running on llama-2-7b.
(VLLM pid=1950315) INFO 07-07 13:43:22 weight_utils.py:218] Using model weights format ['*.safetensors']
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) INFO 07-07 13:43:22 weight_utils.py:218] Using model weights format ['*.safetensors']
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) INFO 07-07 13:43:22 weight_utils.py:218] Using model weights format ['*.safetensors']
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) INFO 07-07 13:43:22 weight_utils.py:218] Using model weights format ['*.safetensors']
(VLLM pid=1950315) INFO 07-07 13:43:27 model_runner.py:255] Loading model weights took 12.8146 GB
(VLLM pid=1950315) (VllmWorkerProcess pid=1950429) INFO 07-07 13:43:27 model_runner.py:255] Loading model weights took 12.8146 GB
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) INFO 07-07 13:43:28 model_runner.py:255] Loading model weights took 12.8146 GB
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) INFO 07-07 13:43:28 model_runner.py:255] Loading model weights took 12.8146 GB
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226] Exception in worker VllmWorkerProcess while processing method determine_num_available_blocks: RuntimeError: Out of workspace memory in AlignedAlloactor, Traceback (most recent call last):
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/executor/multiproc_worker_utils.py", line 223, in _run_worker_process
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     output = executor(*args, **kwargs)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/worker/worker.py", line 173, in determine_num_available_blocks
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self.model_runner.profile_run()
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/worker/model_runner.py", line 874, in profile_run
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self.execute_model(model_input, kv_caches, intermediate_tensors)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/worker/model_runner.py", line 1221, in execute_model
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     model_input.attn_metadata.begin_forward()
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/attention/backends/flashinfer.py", line 132, in begin_forward
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self.prefill_wrapper.begin_forward(
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/flashinfer/prefill.py", line 778, in begin_forward
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self._wrapper.begin_forward(
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226] RuntimeError: RuntimeError: Out of workspace memory in AlignedAlloactor
(VLLM pid=1950315) (VllmWorkerProcess pid=1950430) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226] 
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226] Exception in worker VllmWorkerProcess while processing method determine_num_available_blocks: RuntimeError: Out of workspace memory in AlignedAlloactor, Traceback (most recent call last):
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/executor/multiproc_worker_utils.py", line 223, in _run_worker_process
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     output = executor(*args, **kwargs)
ne 115, in decorate_context
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/worker/worker.py", line 173, in determine_num_available_blocks
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self.model_runner.profile_run()
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/worker/model_runner.py", line 874, in profile_run
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self.execute_model(model_input, kv_caches, intermediate_tensors)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     return func(*args, **kwargs)
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/worker/model_runner.py", line 1221, in execute_model
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     model_input.attn_metadata.begin_forward()
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/vllm/attention/backends/flashinfer.py", line 132, in begin_forward
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self.prefill_wrapper.begin_forward(
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]   File "/home/jingjingl/.conda/envs/harm_project/lib/python3.10/site-packages/flashinfer/prefill.py", line 778, in begin_forward
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226]     self._wrapper.begin_forward(
(VLLM pid=1950315) (VllmWorkerProcess pid=1950428) ERROR 07-07 13:43:36 multiproc_worker_utils.py:226] RuntimeError: RuntimeError: Out of workspace memory in AlignedAlloactor
```

GPU usage when the error happened:
```
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 525.105.17   Driver Version: 525.105.17   CUDA Version: 12.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA RTX A6000    On   | 00000000:01:00.0 Off |                  Off |
| 30%   30C    P8    21W / 300W |  14151MiB / 49140MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   1  NVIDIA RTX A6000    On   | 00000000:25:00.0 Off |                  Off |
| 30%   29C    P8    17W / 300W |  14119MiB / 49140MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   2  NVIDIA RTX A6000    On   | 00000000:41:00.0 Off |                  Off |
| 30%   29C    P8    21W / 300W |  14119MiB / 49140MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
|   3  NVIDIA RTX A6000    On   | 00000000:E1:00.0 Off |                  Off |
| 30%   28C    P8    19W / 300W |  14099MiB / 49140MiB |      0%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+
```

### PR DESCRIPTION
**Pull request by LiuXiaoxuanPKU**
fix https://github.com/flashinfer-ai/flashinfer/issues/362

### CODE REVIEW

**Review by comaniac — APPROVED** [review]
LGTM. Can you add a comment?
Also cc @SolitaryThinker

File: vllm/attention/backends/flashinfer.py [hunk 1]
```diff
@@ -135,11 +135,17 @@ def begin_forward(self):
                 return
 
             assert self.prefill_wrapper is not None
+            assert self.query_start_loc is not None
             assert self.paged_kv_indices is not None
             assert self.paged_kv_indptr is not None
             assert self.paged_kv_last_page_len is not None
-            self.paged_kv_indices = self.paged_kv_indices.to(self.device)
```
Comment (yzh119) [inline vllm/attention/backends/flashinfer.py:141]: Per https://github.com/flashinfer-ai/flashinfer/issues/362#issuecomment-2261713345, perhaps we should keep this?

**Comment by WoosukKwon** [comment]
@LiuXiaoxuanPKU Just for a heads up: we just upgraded the PyTorch and FlashInfer versions in #6951

**Pull request merged**